### PR TITLE
sync to PR100 of intl-numberformat-v3

### PR DESCRIPTION
**Pull request by FrankYFTang**
TC39 2022-07-20 discuss and agree to take 
 https://github.com/tc39/proposal-intl-numberformat-v3/pull/100 
to remove order check

### CODE REVIEW

**Comment by FrankYFTang** [comment]
@sffc @ryanmurakami @romulocintra

**Comment by anba** [comment]
Instead of removing the tests altogether, how about rewriting them to assert that it's now valid that `start` is before the `end` parameter?

**Comment by ptomato** [comment]
Agreed, same comment as on #3609. Otherwise implementations that don't update will still pass the tests.

**Comment by FrankYFTang** [comment]
@anba @ptomato please take another look

**Comment by FrankYFTang** [comment]
PR100 is already landed

File: test/intl402/NumberFormat/prototype/formatRangeToParts/x-greater-than-y-not-throws.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "formatRangeToParts" basic tests when argument  x > y, BigInt included and covers PartitionNumberRangePattern return a object.
+info: |
+  1.1.21 PartitionNumberRangePattern( numberFormat, x, y )
+  (...)
+features: [Intl.NumberFormat-v3]
+---*/
+
+const nf = new Intl.NumberFormat();
+
+// If x > y, throw a RangeError exception.
```
Comment (sffc) [inline test/intl402/NumberFormat/prototype/formatRangeToParts/x-greater-than-y-not-throws.js:16]: should not throw

File: test/intl402/PluralRules/prototype/selectRange/x-greater-than-y-not-throws.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+description: >	
+  "selectRange" basic tests when argument  x > y, return a string.	
+info: |	
+  1.1.6 ResolvePluralRange ( pluralRules, x, y )	
+features: [Intl.NumberFormat-v3]	
+---*/	
+
+const pr = new Intl.PluralRules();	
+
+// 1. If x > y, throw a RangeError exception.
```
Comment (sffc) [inline test/intl402/PluralRules/prototype/selectRange/x-greater-than-y-not-throws.js:15]: should not throw

**Comment by FrankYFTang** [comment]
PTAL

**Review by sffc — DISMISSED** [review]

File: test/intl402/NumberFormat/prototype/formatRange/x-greater-than-y-not-throws.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "formatRange" basic tests when argument  x > y, BigInt included and covers PartitionNumberRangePattern return a string.
+info: |
+  1.1.21 PartitionNumberRangePattern( numberFormat, x, y )
+  (...)
+features: [Intl.NumberFormat-v3]
+---*/
+
+const nf = new Intl.NumberFormat();
+
+// If x > y, throw a RangeError exception.
```
Comment (ptomato) [inline test/intl402/NumberFormat/prototype/formatRange/x-greater-than-y-not-throws.js:16]:
```suggestion
// If x > y, return a string.
```

**Review by ptomato — DISMISSED** [review]
One comment to fix, I'll do it from here

**Review by ptomato — DISMISSED** [review]

**Review by ptomato — APPROVED** [review]

**Comment by ptomato** [comment]
The linter problem was due to trailing spaces in the YAML front matter — we should make it more robust to this, but in the meantime please watch out for trailing spaces!

**Pull request merged**